dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -0,0 +1,97 @@
+require 'acapela'
+require 'net/http'
+require 'uri'
+require 'redcarpet'
+require 'redcarpet/render_strip'
+
+TTS_BUCKET = 'cdo-tts'
+
+# american candidate
+
+# british candidate
+VOICES = {
+  rosie: {
+    VOICE: 'rosie22k',
+    SPEED: 180,
+    SHAPE: 100
+  },
+  ella: {
+    VOICE: 'ella22k',
+    SPEED: 140,
+    SHAPE: 98
+  },
+  sharon: {
+    VOICE: 'sharon22k',
+    SPEED: 180,
+    SHAPE: 100
+  }
+}
+
+module TextToSpeech
+  extend ActiveSupport::Concern
+
+  # TODO: this concern actually depends on the SerializedProperties
+  # concern ... I'm not sure how best to deal with that.
+
+  included do
+    after_save :tts_update
+
+    serialized_attrs %w(
+      tts_instructions_override
+      tts_markdown_instructions_override
+    )
+  end
+
+  def self.tts_upload_to_s3(text, filename, voice=:rosie)
+    return if text.blank?
+    return if AWS::S3.exists_in_bucket(TTS_BUCKET, filename)
+    url = acapela_text_to_audio_url(text, VOICES[voice][:VOICE], VOICES[voice][:SPEED], VOICES[voice][:SHAPE])
+    return if url.nil?
+    uri = URI.parse(url)
+    Net::HTTP.start(uri.host) { |http|
```

Comment:
Super small nit: [prefer `do...end` for multiline blocks](https://github.com/bbatsov/ruby-style-guide#single-line-blocks).